examples/spi-ili9486-esp32-c3/src/main.rs

```diff
@@ -0,0 +1,364 @@
+#![no_std]
+#![no_main]
+
+/* --- Needed by ESP32-c3 --- */
+use esp_backtrace as _;
+use hal::{
+    clock::ClockControl,
+    peripherals::Peripherals,
+    prelude::*,
+    spi::{Spi, SpiMode},
+    timer::TimerGroup,
+    Delay, Rtc, IO,
+};
+/* -------------------------- */
+
+use embedded_graphics::{
+    // Provides the necessary functions to draw on the display
+    draw_target::DrawTarget,
+    // Provides colors from the Rgb565 color space
+    pixelcolor::Rgb565,
+    prelude::RgbColor,
+};
+
+// Provides the parallel port and display interface builders
+use display_interface_spi::SPIInterfaceNoCS;
+
+// Provides the Display builder
+use mipidsi::Builder;
+
+use fugit::RateExtU32;
+
+#[entry]
+fn main() -> ! {
+    let peripherals = Peripherals::take();
+    let mut system = peripherals.SYSTEM.split();
+    let clocks = ClockControl::boot_defaults(system.clock_control).freeze();
+    let io = IO::new(peripherals.GPIO, peripherals.IO_MUX);
+
+    // Disable the RTC and TIMG watchdog timers
+    let mut rtc = Rtc::new(peripherals.RTC_CNTL);
+    let timer_group0 = TimerGroup::new(
+        peripherals.TIMG0,
+        &clocks,
+        &mut system.peripheral_clock_control,
+    );
+    let mut wdt0 = timer_group0.wdt;
+    let timer_group1 = TimerGroup::new(
+        peripherals.TIMG1,
+        &clocks,
+        &mut system.peripheral_clock_control,
+    );
+    let mut wdt1 = timer_group1.wdt;
+    rtc.swd.disable();
+    rtc.rwdt.disable();
+    wdt0.disable();
+    wdt1.disable();
+
+    // Define the delay struct, needed for the display driver
+    let mut delay = Delay::new(&clocks);
+
+    // Define the Data/Command select pin as a digital output
+    let dc = io.pins.gpio7.into_push_pull_output();
+    // Define the reset pin as digital outputs and make it high
+    let mut rst = io.pins.gpio8.into_push_pull_output();
+    rst.set_high().unwrap();
+
+    // Define the SPI pins and create the SPI interface
+    let sck = io.pins.gpio12;
+    let miso = io.pins.gpio11;
+    let mosi = io.pins.gpio13;
+    let cs = io.pins.gpio10;
+    let spi = Spi::new(
+        peripherals.SPI2,
+        sck,
+        mosi,
+        miso,
+        cs,
+        100_u32.kHz(),
+        SpiMode::Mode0,
+        &mut system.peripheral_clock_control,
+        &clocks,
+    );
+
+    // Define the display interface with no chip select
+    let di = SPIInterfaceNoCS::new(spi, dc);
+
+    // Define the display drom the display interface and initialize it
+    let mut display = Builder::ili9486_rgb565(di)
+        .init(&mut delay, Some(rst))
+        .unwrap();
+
+    // Make the display all black
+    display.clear(Rgb565::BLACK).unwrap();
+
+    // Turn SPRITE into an array of pixels in the right colorspace
+    let sprite: [Rgb565; 256] = SPRITE.map(|(r, g, b)| Rgb565::new(r, g, b));
+
+    // Draw the 16*16 sprite on the top left corner of the screen
+    display.set_pixels(0, 0, 15, 15, sprite).unwrap();
```

Comment:
This is pretty inefficient, because the `sprite` array needs to be build in RAM. I'm biased and would simply use embedded-graphics to draw an image, but if you want to demonstrate `set_pixels` I would suggest to change the `SPRITE` constant to be `SPRITE: [Rgb565; 256]` instead.

In addition to that the `SPRITE` constant seems to use values between 0 and 255 while `Rgb565` expects 5 or 6 bit values depending on the channel.

Reply:
I did want to demonstrate how `map` can be used to turn a regular array into an array of RgbXYZ colors, but it does make more sense to just use `embedded-graphics` I will change this.

Reply:
If you want to make this an example of using `set_pixels` I'm OK with that, but the example should make it clear that there are other options available. For more memory constrained targets the array in RAM could be a problem for larger images.

You don't need an array, because `set_pixels` takes an iter: `display.set_pixels(0, 0, 15, 15, SPRITE.map(|(r, g, b)| Rgb565::new(r, g, b))).unwrap();` should also work, without the memory overhead.

Reply:
I opted to use `const SPRITE: [Rgb565; 256] = [Rgb565::RED, Rgb565::ORANGE ...` instead with [`fill_contiguous`](https://docs.rs/embedded-graphics/latest/embedded_graphics/draw_target/trait.DrawTarget.html#method.fill_contiguous). If this is okay you can resolve the conversation. Otherwise do suggest something